lib/Dialect/FIRRTL/Transforms/CreateSiFiveMetadata.cpp

```diff
@@ -85,6 +100,28 @@ LogicalResult CreateSiFiveMetadataPass::emitMemoryMetadata() {
           (mem.numReadPorts() <= 1) && width > 0))
       return;
 
+    SmallDenseMap<Attribute, unsigned> symbolIndices;
+    auto addSymbolToVerbatimOp = [&](Operation *op) -> SmallString<8> {
+      unsigned id;
+      Attribute symbol;
+      if (isa<FModuleLike>(op))
+        symbol = FlatSymbolRefAttr::get((SymbolTable::getSymbolName(op)));
+      else
+        symbol = hw::InnerRefAttr::get(
+            SymbolTable::getSymbolName(op->getParentOfType<FModuleOp>()),
+            getOrAddInnerSym(op));
+      auto it = symbolIndices.find(symbol);
+      if (it != symbolIndices.end()) {
+        id = it->second;
+      } else {
+        id = jsonSymbols.size();
+        jsonSymbols.push_back(symbol);
+        symbolIndices.insert({symbol, id});
+      }
+      SmallString<8> str;
+      ("{{" + Twine(id) + "}}").toVector(str);
+      return str;
+    };
```

Comment:
I'm not sure that this is any better,
```suggestion
    SmallDenseMap<Attribute, unsigned> symbolIndices;
    auto addSymbolToVerbatimOp = [&](Operation *op) -> SmallString<8> {
      Attribute symbol;
      if (auto module = dyn_cast<FModuleLike>(op))
        symbol = FlatSymbolRefAttr::get(module.moduleNameAttr());
      else
        symbol = hw::InnerRefAttr::get(
            op->getParentOfType<FModuleOp>().moduleNameAttr(),
            getOrAddInnerSym(op));
      
      auto [it, inserted] = symbolIndices.try_emplace(symbol, jsonSymbols.size());
      if (inserted)
        jsonSymbols.push_back(symbol);
      
      SmallString<8> str;
      ("{{" + Twine(it->second) + "}}").toVector(str);
      return str;
    };
```